modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-impl/src/main/java/com/liferay/headless/commerce/admin/catalog/internal/resource/v1_0/ProductConfigurationListAccountResourceImpl.java

```diff
@@ -0,0 +1,237 @@
+		}
+
+		return getProductConfigurationListIdProductConfigurationListAccountsPage(
+			cpConfigurationList.getCPConfigurationListId(), null, null,
+			pagination, null);
+	}
+
+	@NestedField(
+		parentClass = ProductConfigurationList.class,
+		value = "productConfigurationListAccounts"
+	)
+	@Override
+	public Page<ProductConfigurationListAccount>
+			getProductConfigurationListIdProductConfigurationListAccountsPage(
+				Long id, String search, Filter filter, Pagination pagination,
+				Sort[] sorts)
+		throws Exception {
+
+		return Page.of(
+			_toProductConfigurationListAccounts(
+				_cpConfigurationListRelService.
+					getAccountEntryCPConfigurationListRels(
+						id, search, pagination.getStartPosition(),
+						pagination.getEndPosition())),
+			pagination,
+			_cpConfigurationListRelService.
+				getAccountEntryCPConfigurationListRelsCount(id, search));
+	}
+
+	@Override
+	public ProductConfigurationListAccount
+			postProductConfigurationListByExternalReferenceCodeProductConfigurationListAccount(
+				String externalReferenceCode,
+				ProductConfigurationListAccount productConfigurationListAccount)
+		throws Exception {
+
+		CPConfigurationList cpConfigurationList =
+			_cpConfigurationListService.
+				fetchCPConfigurationListByExternalReferenceCode(
+					externalReferenceCode, contextCompany.getCompanyId());
+
+		if (cpConfigurationList == null) {
+			throw new NoSuchCPConfigurationListException(
+				"Unable to find product configuration list with external " +
+					"reference code " + externalReferenceCode);
+		}
+
+		return postProductConfigurationListIdProductConfigurationListAccount(
+			cpConfigurationList.getCPConfigurationListId(),
+			productConfigurationListAccount);
+	}
+
+	@Override
+	public ProductConfigurationListAccount
+			postProductConfigurationListIdProductConfigurationListAccount(
+				Long id,
+				ProductConfigurationListAccount productConfigurationListAccount)
+		throws Exception {
+
+		CPConfigurationListRel cpConfigurationListRel =
+			_addCPConfigurationListRel(
+				_cpConfigurationListService.getCPConfigurationList(id),
+				productConfigurationListAccount);
+
+		return _toProductConfigurationListAccount(
+			cpConfigurationListRel.getCPConfigurationListRelId());
+	}
+
+	private CPConfigurationListRel _addCPConfigurationListRel(
+			CPConfigurationList cpConfigurationList,
+			ProductConfigurationListAccount productConfigurationListAccount)
+		throws Exception {
+
+		AccountEntry accountEntry = null;
+
+		if (Validator.isNull(
+				productConfigurationListAccount.
+					getAccountExternalReferenceCode())) {
+
+			accountEntry = _accountEntryService.getAccountEntry(
+				productConfigurationListAccount.getAccountId());
+		}
+		else {
+			accountEntry =
+				_accountEntryService.fetchAccountEntryByExternalReferenceCode(
+					cpConfigurationList.getCompanyId(),
+					productConfigurationListAccount.
+						getAccountExternalReferenceCode());
+
+			if (accountEntry == null) {
```

Comment:
you can buy space by inverting this if, if you first check the erc branch you'll save and indented if

Reply:
just saying because I fear Brian would ask to do this change.. :)